library/src/utils.cxx

```diff
@@ -97,6 +99,14 @@ std::vector<std::string> utils::tokenize(std::string_view str)
   return tokens;
 }
 
+//----------------------------------------------------------------------------
+std::string utils::collapsePath(const std::string& path, const std::string& baseDirectory)
+{
+  return path.empty()       ? path
```

Comment:
weird formatting. clang-format is happy with that?

Reply:
its clang-format doing that, not me ^^